x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx

```diff
@@ -0,0 +1,187 @@
+  attributeName,
+  attributeValue = '',
+  attributes,
+  availableAuthProviders,
+  selectedAuthProviders = [ANY_AUTH_PROVIDER],
+  elasticsearchRoles,
+  disabled,
+  multipleAuthProvidersConfig,
+  handleAttributeSelectorChange,
+  handleAttributeValueChange,
+  handleAuthProviderChange = () => null,
+}) => {
+  return (
+    <EuiPanel
+      data-test-subj="attributeSelector"
+      paddingSize="l"
+      className={disabled ? 'euiPanel--disabled' : ''}
+    >
+      <EuiTitle size="s">
+        <h3>{ATTRIBUTE_SELECTOR_TITLE}</h3>
+      </EuiTitle>
+      <EuiSpacer />
+      {availableAuthProviders && multipleAuthProvidersConfig && (
+        <EuiFlexGroup alignItems="stretch">
+          <EuiFlexItem>
+            <EuiFormRow label={AUTH_PROVIDER_LABEL} fullWidth>
+              <EuiComboBox
+                data-test-subj="authProviderSelect"
+                selectedOptions={getSelectedOptions(selectedAuthProviders, availableAuthProviders)}
+                options={getAuthProviderOptions(availableAuthProviders)}
+                onChange={(options) => {
+                  handleAuthProviderChange(options.map((o) => (o as ChildOption).value));
+                }}
+                fullWidth
+                isDisabled={disabled}
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFlexItem />
+        </EuiFlexGroup>
+      )}
+      <EuiFlexGroup alignItems="stretch">
+        <EuiFlexItem>
+          <EuiFormRow label={EXTERNAL_ATTRIBUTE_LABEL} fullWidth>
+            <EuiSelect
+              name="external-attribute"
+              data-test-subj="ExternalAttributeSelect"
+              value={attributeName}
+              required
+              options={attributes.map((attribute) => ({ value: attribute, text: attribute }))}
+              onChange={(e) => {
+                handleAttributeSelectorChange(e.target.value, elasticsearchRoles[0]);
+              }}
+              fullWidth
+              disabled={disabled}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFormRow label={ATTRIBUTE_VALUE_LABEL} fullWidth>
+            {attributeName === 'role' ? (
+              <EuiSelect
+                value={attributeValue}
+                name="elasticsearch-role"
+                data-test-subj="ElasticsearchRoleSelect"
+                required
+                options={elasticsearchRoles.map((elasticsearchRole) => ({
+                  value: elasticsearchRole,
+                  text: elasticsearchRole,
+                }))}
+                onChange={(e) => {
+                  handleAttributeValueChange(e.target.value);
+                }}
+                fullWidth
+                disabled={disabled}
+              />
+            ) : (
+              <EuiFieldText
+                value={attributeValue}
+                name="attribute-value"
+                placeholder={attributeValueExamples[attributeName as keyof AttributeExamples]}
```

Comment:
I was a little confused by this type and why we were casting here... what do you think of the following:

```diff
diff --git a/x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx b/x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx
index 301815edf28..f1780880b80 100644
--- a/x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx
+++ b/x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx
@@ -32,7 +32,7 @@ import {
 } from './constants';
 
 interface IAttributeSelectorProps {
-  attributeName: string;
+  attributeName: keyof AttributeExamples | 'role';
   attributeValue?: string;
   attributes: string[];
   selectedAuthProviders?: string[];
@@ -51,7 +51,7 @@ interface AttributeExamples {
   metadata: string;
 }
 
-const attributeValueExamples = {
+const attributeValueExamples: AttributeExamples = {
   username: 'elastic,*_system',
   email: 'user@example.com,*@example.org',
   metadata: '{"_reserved": true}',
@@ -173,7 +173,7 @@ export const AttributeSelector: React.FC<IAttributeSelectorProps> = ({
               <EuiFieldText
                 value={attributeValue}
                 name="attribute-value"
-                placeholder={attributeValueExamples[attributeName as keyof AttributeExamples]}
+                placeholder={attributeValueExamples[attributeName]}
                 onChange={(e) => {
                   handleAttributeValueChange(e.target.value);
                 }}

```

Reply:
Much cleaner. Thanks!